transcom/mymove
# Add PPM Documents handler and tests.

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15019) for this change

## Summary

This creates a handler for PPM Documents and connects it to the service object. The endpoint is `/shipments/{shipmentID}/ppm-documents`. Tests check that errors are returned when the SO returns errors, and that the happy path returns a 200 and the documents. This approach is largely based on the existing weight ticket handler.

Note that the YAML definition for Moving Expenses now references `MovingExpenses.yaml` rather than `ProGearWeightTickets`, which I think had been in error.

Note too that the endpoint is listed under PPM in the Swagger UI, though we're querying via the MTO shipment; I assume that listing it here with give it a better chance of being found, since it's not used for non-PPM shipments.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

## Verification Steps for Author

These are to be checked by the author.


## Verification Steps for Reviewers

1. Ensure that the endpoint appears at milmovelocal:3000/swagger-ui/ghc.html or the equivalent, under the PPM heading.
2. Ensure that the payload is as expected.
3. Run server tests.

These are to be checked by a reviewer.



### Backend
